lib/mongodb.js

```diff
@@ -309,6 +309,48 @@ MongoDB.prototype.find = function find(model, id, callback) {
 };
 
 /**
+ * Parses the data input for update operations and returns the
+ * sanitised version of the object.
+ *
+ * @param data
+ * @returns {*}
+ */
+MongoDB.prototype.parseUpdateData = function(model, data) {
+  var parsedData = {};
+
+  if (this.settings.allowExtendedOperators === true) {
+    // Check for other operators and sanitize the data obj
+    var acceptedOperators = [
+      // Field operators
+      '$currentDate', '$inc', '$max', '$min', '$mul', '$rename', '$setOnInsert', '$set', '$unset',
+      // Array operators
+      '$addToSet', '$pop', '$pullAll', '$pull', '$pushAll', '$push',
+      // Bitwise operator
+      '$bit'
+    ];
+
+    var usedOperators = Object.keys(data).filter(function (item) {
+      return acceptedOperators.indexOf(item) !== -1;
+    });
+
+    if (usedOperators.length !== 0) {
+      Object.keys(data).map(function (item) {
+        if ( usedOperators.indexOf(item) === -1 ) {
+          delete data[item];
+        }
+      });
+      parsedData = data;
+    } else {
+      parsedData.$set = data;
+    }
+  } else {
+    parsedData.$set = data;
+  }
+
+  return parsedData;
+};
+
+/**
```

Comment:
Please note the Array filter and map involves closures which have adverse performance impacts. Since this function will be invoked per request, can we optimize them using for-loop?